compiler/circle-mpqsolver/src/pattern/PatternResolver.h

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2023 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __MPQSOLVER_PATTERN_RESOLVER_H__
+#define __MPQSOLVER_PATTERN_RESOLVER_H__
+
+#include <luci/CircleQuantizer.h>
+#include <luci/IR/Module.h>
+
+namespace mpqsolver
+{
+namespace pattern
+{
+
+struct Q8LayerNormWithQ16VarianceResolver
```

Comment:
Could you make a parent class `PatternResolver`?

```cpp
class PatternResolver
{
  virtual ~PatternResolver() = default;
  virtual std::map<luci::CircleNode *, luci::CircleQuantizer::Options::LayerParam> resolver(const luci::Module *module);
}
```

I think it is better to make `resolve` return a map, rather than overwrite `node_params`. A single node can be matched with multiple patterns. The current implementation (overwriting `node_params`) would make it difficult to detect such cases.